src/components/Sidebars/Payback.js

```diff
@@ -162,12 +166,22 @@ const setAllowance = async (
     updateState({ allowanceLoading: true });
 
     const daiToken = maker.getToken('MDAI');
+
     const txPromise = daiToken.approveUnlimited(proxyAddress);
     newTxListener(
       txPromise,
       lang.formatString(lang.transactions.unlocking_token, 'DAI')
     );
-    await txPromise;
+
+    // await maker.service('transactionManager').confirm(txPromise, 1);
+    // - This doesn't work with ganache for some reason?!?
+
+    const _web3 = maker.service('web3');
+    await Promise.all([
+      txPromise,
+      _web3.waitForBlockNumber(_web3.blockNumber() + 2)
+    ]);
+
```

Comment:
This was weird. For whatever reason, the promise to confirm transactions would not resolve, even when spammed with multiple `callGanache('evm_mine');`

Reply:
`callGanache('evm_mine')` has not been 100% reliable in my experience. See if you have better luck using the `mineBlocks` helper, which wraps that in timeouts and sanity checks?